File: lang/nodejs/src/fun/vac/algorithms/sorting/InsertionSort.js
"use strict"

exports.sort = (arr) => {
    let target = null

    let cursor = 0

    for (let i = 1, length = arr.length; i < length; i++) {
        target = arr[i]
        for (cursor = i; cursor > 0; cursor--) {
            if (arr[cursor - 1] > target) {
                arr[cursor] = arr[cursor - 1]
            } else {
                break
            }
        }
        arr[cursor] = target
    }
}

File: lang/python/src/fun/vac/algorithms/sorting/InsertionSort.py
def sort(arr: list) -> None:
    target = None

    cursor = 0

    for i in range(1, len(arr)):
        target = arr[i]
        cursor = i
        while cursor > 0:
            if arr[cursor - 1] > target:
                arr[cursor] = arr[cursor - 1]
            else:
                break
            cursor -= 1
        arr[cursor] = target
